### PR TITLE
Replace session tokens with placeholders in Other Logs page

### PR DESCRIPTION
**Pull request by pandaninjas**
Signed-off-by: pandaninjas <admin@malwarefight.gq>

Split from #381 

This PR only has the session token removal

### CODE REVIEW

**Review by Scrumplex — COMMENTED** [review]
Ideally we should just remove logs and crash reports from the dialog completely. Not sure what would be the best way though.

**Comment by pandaninjas** [comment]
> Ideally we should just remove logs and crash reports from the dialog completely. Not sure what would be the best way though.

Unless we provide another safe way of accessing the logs, people are going to go straight to their .minecraft directory if they cannot access it through Prism, which has zero protection on vanilla as of right now

**Comment by Scrumplex** [comment]
> > Ideally we should just remove logs and crash reports from the dialog completely. Not sure what would be the best way though.
> 
> Unless we provide another safe way of accessing the logs, people are going to go straight to their .minecraft directory if they cannot access it through Prism, which has zero protection on vanilla as of right now

What I meant is that the export dialog shouldn't have the `logs` and `crash-reports` directories at all. Right now when exporting I could only check the logs folder and your PR would just generate an empty zip file.

**Comment by Trial97** [comment]
Now that #1292 was merged this PR can be simplified by having only the "replace session tokens with placeholders" part

**Comment by pandaninjas** [comment]
> Now that #1292 was merged this PR can be simplified by having only the "replace session tokens with placeholders" part

I again forgot that discarding changes to a branch deletes the branch. I'll recreate the PR

File: launcher/ui/pages/instance/OtherLogsPage.cpp [hunk 1]
```diff
@@ -199,6 +200,8 @@ void OtherLogsPage::on_btnReload_clicked()
             showTooBig();
             return;
         }
+         // base64 for {"alg":"HS256"}.{"xuid":"
+        content.replace((QRegularExpression) QRegularExpression("eyJhbGciOiJIUzI1NiJ9\\.eyJ4dWlkIjoi[^\\.]+\\.[A-Za-z0-9]+", QRegularExpression::CaseInsensitiveOption), (QString) "<SESSION_TOKEN>");
```
Comment (Scrumplex) [inline launcher/ui/pages/instance/OtherLogsPage.cpp:204]: do we need the explicit casts?

**Comment by TheKodeToad** [comment]
If the regex was replaced (with a loop or something else) would that (substantially) improve things on large log files? :thinking:
Or would that just be needless complication?

File: launcher/ui/pages/instance/OtherLogsPage.cpp (hunk 1 above)
Comment (Trial97) [inline launcher/ui/pages/instance/OtherLogsPage.cpp:204]: For better performance, I would extract the QRegularExpression as either a static variable or a static member of OtherLogsPage, then in the page constructor [optimize it](https://doc.qt.io/qt-6/qregularexpression.html#optimize), ensuring that the pattern is compiled only once and not every time this function is called.

**Comment by TheKodeToad** [comment]
Open another PR if you want to work on this - or if you still want to work on *this* PR, ask me to reopen. I might give this a try though.

**Pull request closed without merging**